### ISSUE TITLE
New driver release: 495.46

### ISSUE THREAD

**Issue by kmare** [negativo17/nvidia-driver#118]
I hope it's ok to notify about new driver releases here.
A new NVidia driver was just released, with version 495.46. Unfortunately, there is no changelog attached to it, but hopefully it fixes various issues that were introduced in the previous 495.44.

https://forums.developer.nvidia.com/t/linux-solaris-and-freebsd-driver-495-46-new-feature-branch-release/197908

**Comment by scaronni** [comment]
Updating now.

**Comment by adatum** [comment]
Does driver 495.46 work for others? After installing it, Xorg crashes upon trying to log in. 
Reverting back to 495.44 works. 
Fedora 35 MATE.

```
systemd-coredump[3941]: [🡕] Process 3374 (Xorg) of user 0 dumped core.
                                             
                                             Found module linux-vdso.so.1 with build-id: 3a757ae982872004a262a28ed5528bfb98723035
                                             Found module libnss_sss.so.2 with build-id: 63a1597a586dc8670ed177a7f59c4ce5fa54bc34
                                             Found module libffi.so.6 with build-id: 0129f72b58e11d59546f7f207d2c90af7ebd9a5e
                                             Found module libgobject-2.0.so.0 with build-id: ead2bc31b68d342e8a9a6a87148862f08dce1bd1
                                             Found module libgudev-1.0.so.0 with build-id: cea25f33b16ca0c6cd966f48adab0d47accaa2b7
                                             Found module libwacom.so.2 with build-id: a12d81b4a5843f47af7667fdaff121772543164b
                                             Found module libevdev.so.2 with build-id: 46a144442393e4e304aa9e3019fd7a5e93f05a00
                                             Found module libmtdev.so.1 with build-id: d9186323825370faefe916cbb5f7f8458e2aeb5b
                                             Found module libinput.so.10 with build-id: da2f042965357d6ebde46fb16a49bbdf64bed5f2
                                             Found module libinput_drv.so with build-id: 3ea222aad9884e5f1129bfa37b99c1219c523447
                                             Found module libnvidia-glcore.so.495.46 with build-id: 8113b84202e276df12ecc4fc58788506fd5ba386
                                             Found module libnvidia-tls.so.495.46 with build-id: 59f61b3d522665be01a2429d278e7892f4e56e41
                                             Found module libglxserver_nvidia.so with build-id: b448f7b6982718772d814f0b5fd3a01a9c226578
                                             Found module libwfb.so with build-id: df58c33d7b63eeaf02c730560d12624aa5a696d9
                                             Found module libfb.so with build-id: 7ce952b408e5893eb1eccad2b3bf667afcdda6ae
                                             Found module libdl.so.2 with build-id: ba0d0a146dbe192921e9cd581d67b89cb51f9bcf
                                             Found module librt.so.1 with build-id: 08cc365bb9eafa4553df3c707bc6ef2b8fc10a24
                                             Found module nvidia_drv.so with build-id: c0f43c87bd9d909d127cd02f169aace2269ffe5d
                                             Found module libxcb.so.1 with build-id: 1352e33391b3079347f320218799d6e531bd42f1
                                             Found module libGLdispatch.so.0 with build-id: d0cfb761001d324db812228d4bbbd246e8902aff
                                             Found module libXext.so.6 with build-id: 08e82aa07cb7910a08d8386fb16de7896ce290f4
                                             Found module libX11.so.6 with build-id: b63e8f330442522b1d5872d1b2ccf32702767d80
                                             Found module libGLX.so.0 with build-id: de2a892b9cce65d9a9a554496091a598da274ef7
                                             Found module libGL.so.1 with build-id: 7314905a994a36ab97accfd0d57d9fa6b2151ca0
                                             Found module libglx.so with build-id: d3cd137b5a66683cb2a748101f6089e5bc40b5db
                                             Found module libpcre.so.1 with build-id: fcbdb4731e50f20a47b62341a6841dd616fe495d
                                             Found module libbrotlicommon.so.1 with build-id: 9e92a8ec2d9efe80ec86e04fba0549cd3202ebc4
                                             Found module libgraphite2.so.3 with build-id: 58f949a538f65a6221e442879144b368f663f7e3
                                             Found module libglib-2.0.so.0 with build-id: 3720108994016259cac8f61f174ea32513518152
                                             Found module libgpg-error.so.0 with build-id: 081975d0a3416374b4883b2f1639fd3c9df42390
                                             Found module libbrotlidec.so.1 with build-id: 57cb8a53e33f31a620f3739919414c0d00549f87
                                             Found module libharfbuzz.so.0 with build-id: 620a4b6e2914d6e1301c086def09a60e58450dc3
                                             Found module libpng16.so.16 with build-id: afcd45d29b0d579615db44a29213544328adde6f
                                             Found module libbz2.so.1 with build-id: 19ce300b78962ee0b113afb0f1016727a2f3c37b
                                             Found module libcap-ng.so.0 with build-id: 6688fe6b1ee5a2fc3c1ec7a8507b02cacb06c236
                                             Found module libgcrypt.so.20 with build-id: 711d41580c5a8649a79a8430a985dac3e25b5ba2
                                             Found module libcap.so.2 with build-id: 0214aa9cc6a8646eb9ec27ab7dda6a2219da6500
                                             Found module liblz4.so.1 with build-id: fd02c4542a1ce1ad6c503b958447be1c37a4afee
                                             Found module libzstd.so.1 with build-id: c594c701d8f5ba174a5be306973fa9803d65e327
                                             Found module liblzma.so.5 with build-id: 7fec53ce7cba9489f130c99cfd3ace82e9dde0ee
                                             Found module libfreetype.so.6 with build-id: 6d98d390da279e5c22ebe472a552b3470554655b
                                             Found module libfontenc.so.1 with build-id: 14821e9de937cc8596b6de5be8340195ba723f8b
                                             Found module libz.so.1 with build-id: 5903f5c355c264403e4e7cdc66779584425ca3b8
                                             Found module libpcre2-8.so.0 with build-id: dbf58040a44903b428c4e68716aa4f45058ad650
                                             Found module ld-linux-x86-64.so.2 with build-id: b43118df1fdb4c0aff150b6f8f926bccdec2a7f0
                                             Found module libgcc_s.so.1 with build-id: 88564abce789aa42536da1247a57ff6062d61dcb
                                             Found module libc.so.6 with build-id: 644dac2c66a6e0b32674f0ec71e7431bd0c06a63
                                             Found module libm.so.6 with build-id: cea55efc551e62cd8439d044aa9a765c445fb7bf
                                             Found module libaudit.so.1 with build-id: f5aa484ba17b5425d516f2b879d7181e3adb659d
                                             Found module libXdmcp.so.6 with build-id: 931df0f14ee003f06c94f5c1ea5e89de2872d8b5
                                             Found module libxshmfence.so.1 with build-id: 12db5b6f0c14db1bed5d99bcb53a0e4aaf2bda51
                                             Found module libsystemd.so.0 with build-id: e4996d7fd9705035bdd6072e68c9dc65647d7d1c
                                             Found module libXau.so.6 with build-id: b6a2d4859848c28bf0a4fc8744d04b1935879af6
                                             Found module libXfont2.so.2 with build-id: 58a389854dac0aa5739392f60e1f815e5e711389
                                             Found module libpixman-1.so.0 with build-id: dc521afa5e1d55834eebac7bb3c97cfe5e73f384
                                             Found module libdrm.so.2 with build-id: fc68e955c370337e29016e39ba13d13bd90f9b25
                                             Found module libpciaccess.so.0 with build-id: 3595c5846a56f550736a0b054f3f17522fba7f9a
                                             Found module libunwind.so.8 with build-id: 656a76d5a68902c2d0ace44c84a4840cffdaa675
                                             Found module libcrypto.so.1.1 with build-id: 3ead607210606611d5f3398ee7e65198bd519de9
                                             Found module libselinux.so.1 with build-id: f805394f993c704b949315b56c344d22dfad801f
                                             Found module libudev.so.1 with build-id: 7422938010fad35c79a9a6cb017799a1cc575e7f
                                             Found module libdbus-1.so.3 with build-id: 58e800f0865bde326c5c9ce510062d5b19902f67
                                             Found module Xorg with build-id: 8d6dac7ceaac6e66e8752a623031c741ab40ae96
                                             Stack trace of thread 3374:
                                             #0  0x00007fd7a516084c __pthread_kill_implementation (libc.so.6 + 0x8f84c)
                                             #1  0x00007fd7a51136a6 raise (libc.so.6 + 0x426a6)
                                             #2  0x00007fd7a50fd7d3 abort (libc.so.6 + 0x2c7d3)
                                             #3  0x0000555dcc2891a0 OsAbort (Xorg + 0x1ba1a0)
                                             #4  0x0000555dcc291476 FatalError (Xorg + 0x1c2476)
                                             #5  0x0000555dcc287d6a OsSigHandler (Xorg + 0x1b8d6a)
                                             #6  0x00007fd7a5113750 n/a (libc.so.6 + 0x42750)
                                             
                                             Stack trace of thread 3544:
                                             #0  0x00007fd7a51e2c1e epoll_wait (libc.so.6 + 0x111c1e)
                                             #1  0x0000555dcc28ac58 ospoll_wait (Xorg + 0x1bbc58)
                                             #2  0x0000555dcc28adc0 InputThreadDoWork (Xorg + 0x1bbdc0)
                                             #3  0x00007fd7a515ea87 start_thread (libc.so.6 + 0x8da87)
                                             #4  0x00007fd7a51e3640 __clone3 (libc.so.6 + 0x112640)
```

```
Packages Altered:
    Upgrade  akmod-nvidia-3:495.46-1.fc35.x86_64            @fedora-nvidia
    Upgraded akmod-nvidia-3:495.44-1.fc35.x86_64            @@System
    Upgrade  nvidia-driver-3:495.46-1.fc35.x86_64           @fedora-nvidia
    Upgraded nvidia-driver-3:495.44-1.fc35.x86_64           @@System
    Upgrade  nvidia-driver-NVML-3:495.46-1.fc35.x86_64      @fedora-nvidia
    Upgraded nvidia-driver-NVML-3:495.44-1.fc35.x86_64      @@System
    Upgrade  nvidia-driver-cuda-3:495.46-1.fc35.x86_64      @fedora-nvidia
    Upgraded nvidia-driver-cuda-3:495.44-1.fc35.x86_64      @@System
    Upgrade  nvidia-driver-cuda-libs-3:495.46-1.fc35.x86_64 @fedora-nvidia
    Upgraded nvidia-driver-cuda-libs-3:495.44-1.fc35.x86_64 @@System
    Upgrade  nvidia-driver-libs-3:495.46-1.fc35.i686        @fedora-nvidia
    Upgraded nvidia-driver-libs-3:495.44-1.fc35.i686        @@System
    Upgrade  nvidia-driver-libs-3:495.46-1.fc35.x86_64      @fedora-nvidia
    Upgraded nvidia-driver-libs-3:495.44-1.fc35.x86_64      @@System
    Upgrade  nvidia-kmod-common-3:495.46-1.fc35.noarch      @fedora-nvidia
    Upgraded nvidia-kmod-common-3:495.44-3.fc35.noarch      @@System
    Upgrade  nvidia-libXNVCtrl-3:495.46-1.fc35.x86_64       @fedora-nvidia
    Upgraded nvidia-libXNVCtrl-3:495.44-1.fc35.x86_64       @@System
    Upgrade  nvidia-libXNVCtrl-devel-3:495.46-1.fc35.x86_64 @fedora-nvidia
    Upgraded nvidia-libXNVCtrl-devel-3:495.44-1.fc35.x86_64 @@System
    Upgrade  nvidia-persistenced-3:495.46-1.fc35.x86_64     @fedora-nvidia
    Upgraded nvidia-persistenced-3:495.44-1.fc35.x86_64     @@System
    Upgrade  nvidia-settings-3:495.46-1.fc35.x86_64         @fedora-nvidia
    Upgraded nvidia-settings-3:495.44-1.fc35.x86_64         @@System
```

**Comment by adatum** [comment]
I noticed on the first reboot after installing 495.46, the system gets stuck here in the boot process:

![IMG_20211214_225313](https://user-images.githubusercontent.com/9773655/146124075-0a92358d-23d2-42ad-95ae-096dbc1a42e6.jpg)

After a Ctrl+Alt+Del to reboot again, it doesn't get stuck and makes it to lightdm, but logging in always fails with Xorg crashing, as shown in my previous post, and returns to the lightdm login screen. 

I tried the 495.46 installation again after upgrading to kernel 5.15.7-200.fc35.x86_64, with the same results.

**Comment by scaronni** [comment]
Weird, it's working fine on all my systems. I'll make some more tests.

**Comment by scaronni** [comment]
> I upgraded to 495.46 as well, and while it looked to be all good, as soon as I tried to run some games I kept getting errors like: SDL couldn't find matching GLX visual. Browsers seemed to be ok though, glxinfo -B showing the proper info etc... Reverting to 495.44 fixed the issue of course. Might be an issue with the 495.46 version?

Are you using a spin like Mate as well?

**Comment by kmare** [comment]
@scaronni no, I use plain/vanilla Gnome on X11.
From the few tests I've done, the flathub related games/apps are having the issue. Eg.
https://flathub.org/apps/details/org.armagetronad.ArmagetronAdvanced
https://flathub.org/apps/details/org.openmw.OpenMW

**Comment by scaronni** [comment]
And not steam/other ones?

**Comment by kmare** [comment]
I can't validate yet for steam/other until tonight (don't really have anything steam-related installed). I'll test it out and let you know.
As I mentioned before, browsers (Firefox and Chrome) seem to be working fine with eg. WebGL.

**Comment by scaronni** [comment]
Everything seems to be working fine on my systems, also with games and what not. So, I can't reproduce.

**Comment by kmare** [comment]
@scaronni thank you for all the testing and hard work!
I wanted to report that it now works fine after re-installing the drivers (updated the kernel to 5.15.7 as well).
I also updated the flaptak nvidia libraries/drivers (org.freedesktop.Platform.GL.nvidia-495-46).
Not sure which of the 2 actually did it, but everything is working properly now.
Thank you again.

**Comment by scaronni** [comment]
I don't know what the flatpak nvidia gl thing is, but for sure if libraries and other driver components don't match, it will for sure not work.

**Comment by kmare** [comment]
Yup, of course. I'm not sure what this flatpak nvidia platform really is, my guess is that it's being pulled by flatpak apps.

**Comment by adatum** [comment]
I tried installing 495.46 again this time with kernel 5.15.8 and still the same issue with Xorg crashing and lightdm failing to start properly. Is there anything I can check? 

I've never had an issue like this before. Undoing the `dnf` transaction and going back to 495.44 works as usual. 

I'm not using flatpaks, etc just a regular installation with the system package manager.

**Comment by scaronni** [comment]
Can you try booting directly in GDM and not lightdm as a test? Are you on an Optimus system?

Anyway the regression is not on the packaging side, as you can see nothing has changed: https://github.com/negativo17/nvidia-driver/commit/bb7292c4f638e4fa78122c7096b846be9d0b2eb3

You should probably open a bug to Nvidia.

**Comment by adatum** [comment]
You're probably right that it's an nvidia issue. Others are reporting exactly the same problem with 495.46: https://forums.developer.nvidia.com/t/x-server-1-21-1-1-crash-on-startx-with-nvidia-driver-495-46/198417

It's not clear what is the common factor among those who have this problem, and also why others don't.

I'm not on an Optimus system (it's a desktop) and I don't have GDM installed. I'm running Fedora 35 MATE which uses lightdm by default.

**Comment by leigh123linux** [comment]
> Can you try booting directly in GDM and not lightdm as a test? Are you on an Optimus system?
> 
> Anyway the regression is not on the packaging side, as you can see nothing has changed: [bb7292c](https://github.com/negativo17/nvidia-driver/commit/bb7292c4f638e4fa78122c7096b846be9d0b2eb3)
> 
> You should probably open a bug to Nvidia.

lightdm is working fine here.

**Comment by adatum** [comment]
The issue is due to a bug in xserver: https://gitlab.freedesktop.org/xorg/xserver/-/issues/1275
In Fedora, it has been addressed by https://bugzilla.redhat.com/show_bug.cgi?id=2046147 and solved with xorg-x11-server-1.20.14-3 https://bodhi.fedoraproject.org/updates/FEDORA-2022-a2017f20aa
